ark/phenotyping/run_cell_som.R

```diff
@@ -13,48 +13,53 @@ library(FlowSOM)
 # get the command line arguments
 args <- commandArgs(trailingOnly=TRUE)
 
-# get the path to the cluster counts data
+# get the path to the cluster counts data (norm)
 clusterCountsPath <- args[1]
 
 # get the weights write path
 cellWeightsPath <- args[2]
 
-# get the cluster write path
-cellClusterPath <- args[3]
+# get the cluster write path (normalized)
+cellClusterPathNorm <- args[3]
 
-# read the cluster counts data
+# read the cluster counts data (norm)
 print("Reading the cluster counts data")
-clusterCountsData <- arrow::read_feather(clusterCountsPath)
+clusterCountsNorm <- as.data.frame(arrow::read_feather(clusterCountsPath))
 
 # read the weights
 print("Reading the weights matrix")
 somWeights <- as.matrix(arrow::read_feather(cellWeightsPath))
 
-# get the column names of the SOM pixel/meta clusters
-clusterCols <- colnames(clusterCountsData)[grepl("cluster_|hCluster_cap_",
-                                           colnames(clusterCountsData))]
+# define the subset of count columns to cluster on
+clusterCols <- colnames(clusterCountsNorm)[grepl("pixel_som_cluster_|pixel_meta_cluster_",
+                                           colnames(clusterCountsNorm))]
 
-# keep just the cluster columns
-clusterCountsNorm <- clusterCountsData[, clusterCols]
+# subset on just the clusterCols for normalization
+clusterCountsNormSub <- as.matrix(clusterCountsNorm[,clusterCols])
```

Comment:
Just wondering, is there a reason you're saving this to clusterCountsNormSub here? In create_cell_som.R, you have this line as "clusterCountsNorm <- as.matrix(clusterCountsNorm[,clusterCols])". Not a big deal since this is obviously still correct, but I'm just wondering if I'm missing something.

Reply:
The only reason it's called `clusterCountsNormSub` is because I've subsetted on just the `clusterCols`, which are the set of columns defining the pixel cluster counts. Just a way to distinguish between the full `clusterCountsNorm` dataset which contains info about the segmentation labels and cell sizes.

Reply:
I get that, but my question is why do it here and not in create_cell_som.R? It's the same operation being done.

Reply:
OK I see, I can make that change.